File: solve/03_로또의 최고 순위와 최저 순위 copy/postforty/로또의 최고 순위와 최저 순위.py
def solution(lottos, win_nums):
    answer = []

    count = 0
    zeroCount = 0

    for i in lottos:
        if i in win_nums:
            count += 1
        if i == 0:
            zeroCount += 1

    level = lambda x : 1 if x == 6 else 2 if x == 5 else 3 if x == 4 else 4 if x == 3 else 5 if x == 2 else 6
            
    answer = [level(count+zeroCount), level(count)]
    
    return answer

print(solution([44, 1, 0, 0, 31, 25], [31, 10, 45, 1, 6, 19]))

File: solve/0429_로또의 최고 순위와 최저 순위/postforty/로또의 최고 순위와 최저 순위.js
function solution(lottos, win_nums) {
  var answer = [];
  let count = 0;
  let zeroCount = 0;

  for (let i of lottos) {
    if (win_nums.includes(i)) {
      count++;
    }
    if (i === 0) {
      zeroCount++;
    }
  }

  const level = (value) =>
    value === 6
      ? 1
      : value === 5
      ? 2
      : value === 4
      ? 3
      : value === 3
      ? 4
      : value === 2
      ? 5
      : 6;

  answer = [level(count + zeroCount), level(count)];
  return answer;
}

console.log(solution([44, 1, 0, 0, 31, 25], [31, 10, 45, 1, 6, 19]));
